ntoskrnl/io/iomgr/irq.c

```diff
@@ -153,25 +155,22 @@ IoDisconnectInterrupt(PKINTERRUPT InterruptObject)
     KeDisconnectInterrupt(&IoInterrupt->FirstInterrupt);
 
     /* Now disconnect the others */
-    for (i = 0; i < KeNumberProcessors; i++)
+    for (i = 0; i < _countof(IoInterrupt->Interrupt) && IoInterrupt->Interrupt[i]; ++i)
```

Comment:
This change would imply that you would stop the for-loop if you have a NULL-pointer somewhere inside your array, even if there are valid ones that follow next.
Is this what you want to do?

Reply:
Yes, I do: that's the 'Assign other interrupts contiguously' commit.